src/relay/transforms/pattern_utils.h

```diff
@@ -441,7 +441,7 @@ static inline long double ToScalar(const runtime::NDArray& array, size_t i = 0)
 static inline Array<Integer> ToVector(const runtime::NDArray& array) {
   size_t ndim = array.Shape().size();
   ICHECK_EQ(ndim, 1) << "This function should only be used for 1D NDArrays";
-  size_t len = array.Shape().front();
+  size_t len = array.Shape().at(0);
```

Comment:
let's add `.front()` and `.back()` like what is done in `runtime::Array`

Reply:
not sure whether we need those for a tuple object

Reply:
Several lines of code that are good for future development. @zhiics asked me to add them to `runtime::Array` last time, and it turns out to be useful right now :-)